File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/VampireRing.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class VampireRing extends Equipable {
	{
		description = Locale.get("vampire_ring_desc");
		name = Locale.get("vampire_ring");
		sprite = "item-ring_j";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).vampire += 5f;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).vampire -= 5f;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/VampireRing.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equipable

import org.rexcellentgames.burningknight.assets.Locale

class VampireRing : Equipable() {
	init {
		description = Locale.get("vampire_ring_desc")
		name = Locale.get("vampire_ring")
		sprite = "item-ring_j"
	}

	override fun onEquip() {
		super.onEquip()
		this.owner.vampire += getChance()
	}

	override fun getMaxLevel(): Int {
		return 10
	}

	override fun onUnequip() {
		super.onUnequip()
		this.owner.vampire -= getChance()
	}

	private fun getChance(): Float {
		return this.level * 10f
	}

	override fun getDescription(): String {
		return super.getDescription().replace("{CHANCE}", getChance().toInt().toString())
	}
}